### PR TITLE
DSDEGP-1370: update SearchDAO to return Document instead of MappedOutput

### PR DESCRIPTION
**Pull request by pshapiro4broad**
- the caller must do the mapping now
- also updated DAO to accept QueryDefinition directly, so the caller must now map query to QueryDefinition

This is preparation for the work in DSDEGP-1370.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala [hunk 1]
```diff
@@ -36,7 +33,7 @@ trait SearchDAO {
     * @param document A (potentially partial) metadata document containing
     *                 new fields to set on the document in the index.
     * @param index    The index in which to update the document.
-    * @tparam D       The type of the document.
+    * @tparam D The type of the document.
```
Comment (danxmoran) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala]: Indentation lost here, not sure if we care.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala [hunk 2]
```diff
@@ -45,18 +42,10 @@ trait SearchDAO {
...
-    * @param input       An object describing the query to run.
+    * @param queryDefinition       The query to run.
     * @param index       The index to run the query against.
-    * @param queryMapper Utility for mapping the query input to an
-    *                    elastic4s query object, and for mapping the
-    *                    query results back to a query output type.
-    * @tparam I          The type of the query input.
-    * @tparam O          The type of the query output.
-    * @tparam D          The type of the document to query.
+    * @tparam D The type of the document to query.
```
Comment (danxmoran) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala]: Also here.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/SearchService.scala [hunk 3]
```diff
@@ -31,8 +30,12 @@ class SearchService private (searchDAO: SearchDAO) {
     transferInput: TI,
     index: ElasticsearchIndex[D],
     queryMapper: ElasticsearchQueryMapper[TI, TO, D]
-  ): Future[Seq[TO]] = {
-    searchDAO.queryMetadata(transferInput, index, queryMapper)
+  )(implicit ec: ExecutionContext): Future[Seq[TO]] = {
+    if (queryMapper.isEmpty(transferInput)) {
+      Future.successful(Seq.empty)
+    } else {
+      searchDAO.queryMetadata(queryMapper.buildQuery(transferInput), index) map queryMapper.toQueryOutputs
```
Comment (danxmoran) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/service/SearchService.scala]: Nitpick: I don't think we use much infix notation, could you instead do `searchDAO.queryMetadata(...).map(queryMapper.toQueryOutputs)`?

**Review by danxmoran — APPROVED** [review]

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala (hunk 2 above)
Comment (pshapiro4broad) [reply clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/SearchDAO.scala]: Personally I think lining these up is a waste of time, but I may as well preserve the current file's convention.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/SearchService.scala (hunk 3 above)
Comment (pshapiro4broad) [reply clio-server/src/main/scala/org/broadinstitute/clio/server/service/SearchService.scala]: OK

**Review by tbl3rd — APPROVED** [review]

**Pull request merged**